src/app/content/hooks/locationChange/resolveContent.ts

```diff
@@ -219,8 +224,8 @@ const loadContentReference = async(
       page: getUrlParamForPageId(targetBook, reference.pageUid),
     },
     state: {
-      bookUid: targetBook.id,
-      bookVersion: targetBook.version,
+      bookUid: reference.bookId || targetBook.id,
```

Comment:
Instead of conditions here I think what you want to do is use your reference input to conditionally bypass the target book logic above so that it doesn't make that extra query